### ISSUE TITLE
Test 'Authentication: client max_int chain, server required' fails on OS X with all options enabled

### ISSUE THREAD

**Issue by redtangent** [ARMmbed/mbedtls#1874]
### Description
- Type: Bug
- Priority: Major

---------------------------------------------------------------
## Bug

**OS**  
OSX

**mbed TLS build:**  
Version: 7d728bd70e6a
Configuration: All options but `MBEDTLS_MEMORY_BACKTRACE` enabled 

When the full `config.h` options are enabled on an OS X build, the test 'Authentication: client max_int chain, server required' fails because of a memory allocation problem.

I think this has been caused because additional features have grown the memory footprint. Disabling the option `MBEDTLS_CHACHAPOLY_C`alone, allows the test to work again.

Curiously, this impacts OS X but not Linux.

To reproduce:
```
git checkout 7d728bd70e6a8d129ee07813da1538de1bf95cb2
./scripts/config.pl full
./scripts/config.pl unset MBEDTLS_MEMORY_BACKTRACE
make test; make programs
./tests/ssl-opt.sh -f 'Authentication: client max_int chain, server required'
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-2415

**Comment by sbutcher-arm** [comment]
The issue has been fixed by PR #1875 which has now been merged, so this issue can be closed.
